From 1c1d595cedb882fe0b5574b24c07fb34c84e5a9d Mon Sep 17 00:00:00 2001
From: Robert Mader <robert.mader@collabora.com>
Date: Wed, 28 Jan 2026 15:46:06 +0100
Subject: [PATCH 6/6] aperture: Prefer openh264 over x264

The former performs significantly better, especially on mobile devices.
While x264 is expected to be the better encoder overall, it's not
totally unexpected that openh264 performs better in the use case at
hand, given it's strong focus on webrtc / calls. Possibly x264 could
archive similar performance with the appropriate options.

This essentially unbreaks video recording on mobile devices such as the
FairPhone 5.
---
 aperture/src/viewfinder.rs | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/aperture/src/viewfinder.rs b/aperture/src/viewfinder.rs
index 64dfcd0..0cdc0f5 100644
--- a/aperture/src/viewfinder.rs
+++ b/aperture/src/viewfinder.rs
@@ -993,6 +993,12 @@ impl Viewfinder {
                     }
                     hw_encoder_found = true;
                 }
+                if let Some(encoder) = registry.lookup_feature("openh264enc") {
+                    encoder.set_rank(gst::Rank::PRIMARY);
+                }
+                if let Some(encoder) = registry.lookup_feature("x264enc") {
+                    encoder.set_rank(gst::Rank::PRIMARY - 1);
+                }
                 log::debug!(
                     "Setting up recording with h264/mp4 profile {} hw acceleration",
                     if self.enable_hw_encoding() && hw_encoder_found {
-- 
2.52.0

